datafusion/core/src/datasource/file_format/file_type.rs

```diff
@@ -111,6 +115,58 @@ impl FileCompressionType {
         self.variant.is_compressed()
     }
 
+    /// Given a `Stream`, create a `Stream` which data are decompressed with `FileCompressionType`.
+    pub fn convert_to_compress_stream<
+        T: Stream<Item = Result<Bytes>> + Unpin + Send + 'static,
+    >(
+        &self,
+        s: T,
+    ) -> Result<Box<dyn Stream<Item = Result<Bytes>> + Send + Unpin>> {
+        #[cfg(feature = "compression")]
+        let err_converter = |e: std::io::Error| match e
+            .get_ref()
+            .and_then(|e| e.downcast_ref::<DataFusionError>())
+        {
+            Some(_) => {
+                *(e.into_inner()
+                    .unwrap()
+                    .downcast::<DataFusionError>()
+                    .unwrap())
```

Comment:
safe unwraps?

Reply:
I think it's fine.
Because err_converter only accepts std::io::Error type as input, and DataFusionError has also implemented it accordingly.

https://github.com/apache/arrow-datafusion/blob/fe46a1ed9833f2f9ea4c4ccd4d77718e5c371ab1/datafusion-common/src/error.rs#L78-L82